pkg/api/types.go

```diff
@@ -1407,6 +1409,69 @@ type ReplicationControllerList struct {
 	Items []ReplicationController `json:"items"`
 }
 
+// DisruptionBudget is an object to define the max disruption that can be caused to a collection of pods
+type DisruptionBudget struct {
+	// Specification of the desired behavior of the DisruptionBudget.
+	Spec DisruptionBudgetSpec `json:"spec,omitempty"`
+	// Most recently observed status of the DisruptionBudget.
+	Status DisruptionBudgetStatus `json:"status,omitempty"`
+}
+
+// DisruptionBudgetSpec is a description of a DisruptionBudget.
+type DisruptionBudgetSpec struct {
+	// RateLimit is a rate limit on disruptions (preemption and other evictions) across the corresponding
+	// collection of pods, e.g. no more than one disruption per hour across the pods of a particular Service
+	RateLimit *RateLimitSpec `json:"ratelimit,omitempty"`
+	// The maximum number of pods that can be unavailable simultaneously.
+	// Value can be an absolute number (ex: 5) or a percentage of total pods(ex: 10%).
+	// Absolute number is calculated from percentage by rounding up.
+	// By default, a fixed value of 1 is used.
+	MaxUnavailable intstr.IntOrString `json:"maxUnavailable,omitempty"`
```

Comment:
I think that users generally want to specify MinAvailable.  They might seem like the same thing, but they're not.

Suppose a pod is evicted and its replacement has been sent to the kubelet, but it's not yet in ready state.  If MinReadySeconds haven't yet elapsed, one view is that the new pod isn't *yet* unavailable.

OTOH, no one would say the pod is available.  It will be less confusing to specify MinAvailable.